captum/optim/_utils/models.py

```diff
@@ -0,0 +1,98 @@
+from typing import List
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+def get_model_layers(model) -> List[str]:
+    """
+    Return a list of hookable layers for the target model.
+    """
+    layers = []
+
+    def get_layers(net, prefix: List = []) -> None:
+        if hasattr(net, "_modules"):
+            for name, layer in net._modules.items():
+                if layer is None:
+                    continue
+                layers.append(".".join(prefix + [name]))
+                get_layers(layer, prefix=prefix + [name])
+
+    get_layers(model)
+    return layers
+
+
+class RedirectedReLU(torch.autograd.Function):
+    """
+    A workaround when there is no gradient flow from an initial random input.

```

Comment:
The PyTorch version of RedirectedReLU was originally written by @greentfrapp and it a less sophisticated version of Lucid's RedirectedReLU. For the moment @greentfrapp's version is technically "wrong", but it functions well enough to be used for the moment. Without RedirectedReLU, many of the channels and neurons will never render at all.
 
https://github.com/greentfrapp/lucent/blob/master/lucent/modelzoo/inceptionv1/helper_layers.py#L61-L77

https://github.com/tensorflow/lucid/blob/master/lucid/optvis/overrides/redirected_relu_grad.py

Reply:
I've added a description about what it does and why it's needed.

Reply:
ReLU layers will block the gradient flow during backpropagation when their input is less than 0. This means that it can be impossible to visualize a target without allowing negative values to pass through ReLU layers during backpropagation.